torch/csrc/api/src/nn/modules/activation.cpp

```diff
@@ -96,5 +96,20 @@ void LogSigmoidImpl::pretty_print(std::ostream& stream) const {
   stream << "torch::nn::LogSigmoid()";
 }
 
+// ============================================================================
+
+SoftmaxImpl::SoftmaxImpl(const SoftmaxOptions& options_)
+    : options(options_) {}
+
+void SoftmaxImpl::reset() {}
+
+void SoftmaxImpl::pretty_print(std::ostream& stream) const {
+  stream << "torch::nn::Softmax(dim=" << options.dim() << ")";
+}
+
+Tensor SoftmaxImpl::forward(const Tensor& input) {
+  return input.softmax(options.dim());
```

Comment:
From [Implementation Notes](https://github.com/pytorch/pytorch/issues/25883):
> You might see that some of the torch.nn modules call the corresponding torch.nn.functional functions. When implementing the C++ version of those modules, we should also add the corresponding torch::nn::functional functions, in order to preserve the call structure.

Please port [this](https://github.com/pytorch/pytorch/blob/master/torch/nn/functional.py#L1202-L1234) logic to [torch/csrc/api/include/torch/nn/functional/activation.h](https://github.com/pytorch/pytorch/blob/master/torch/csrc/api/include/torch/nn/functional/activation.h) and test it in [test/cpp/api/functional.cpp](https://github.com/pytorch/pytorch/blob/master/test/cpp/api/functional.cpp)

Reply:
Thanks. I ported python logic to functional/activation.h and tested it.